static/app/views/settings/featureFlags/useUserFromId.tsx

```diff
@@ -0,0 +1,19 @@
+import type {User} from '@sentry/types';
+
+import {useApiQuery} from 'sentry/utils/queryClient';
+import useOrganization from 'sentry/utils/useOrganization';
+
+export default function UseUserFromId({id}: {id: number}) {
+  const organization = useOrganization();
+
+  const {isPending, isError, data} = useApiQuery<User>(
+    [`/organizations/${organization.slug}/users/${id}/`],
+    {staleTime: 0}
+  );
+
+  if (isError) {
```

Comment:
```suggestion

  if (typeof id === 'string') {
    return id;
  }

  if (isError) {
```

Reply:
hmm, i think we can just update this when we make the change to user? we wouldn't even need the hook then. if we change to username i'll just update it when that happens too

Reply:
ya, do this one later

what'll happen though is 3 PRs will land: 
- frontend gets ready for new api response shape, can handle old and new, is deployed everywhere
- api chnage is deployed everywhere
- frontend drops support for old number-only `id` and just works with the new response shape